src/app/services/window.service.ts

```diff
@@ -135,41 +164,14 @@ export class WindowService {
       if (!data.version || !data.type || data.type !== 'window') {
         throw new Error('File is not a valid Altair file.');
       }
+
       // Importing window data...
       // Add new window
-      // Set window name
-      // Set API URL
-      // Set query
-      // Set headers
-      // Set variables
-      // Set subscription URL
+      // Set required data
       this.newWindow().subscribe(newWindow => {
         const windowId = newWindow.windowId;
 
-        if (data.windowName) {
```

Comment:
You still have the `populateNewWindow()` method separated but looking at the `importWindowData()` method, that is exactly what you need. So `populateNewWindow()` becomes counterintuitive 😄, especially since

```js
this.newWindow().subscribe(newWindow => {
	       this.newWindow().subscribe(newWindow => {
         const windowId = newWindow.windowId;
        this.populateNewWindow(windowId, data);
```

is being repeated multiple times